### PR TITLE
Adds documentation to api client public methods

### PR DESCRIPTION
**Pull request by alonsosalasinfante**

### CODE REVIEW

File: lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb [hunk 1]
```diff
@@ -102,6 +102,14 @@ def get_upload_token(app_id, binary_path)
         return upload_token_format(response.body[:appId], response.body[:projectNumber], binary_hash)
       end
 
+      # Uploads the app binary to the Firebase API
+      #
+      # args
+      #   app_id - Firebase App ID
+      #   binary_path - Absolute path to your app's apk/ipa file
+      #   platform - android vs ios
```
Comment (lfkellogg) [inline lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb:110]: how about `'android' or 'ios'` to make it clear that one of two string values is expected

File: lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb [hunk 2]
```diff
@@ -160,14 +168,22 @@ def upload(app_id, binary_path, platform)
         upload_status_response.release_id
       end
 
-      # Gets the upload status for the app release.
-      def get_upload_status(app_id, app_token)
+      # Gets the upload_status for this
```
Comment (lfkellogg) [inline lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb:171]: "Fetches the status of an uploaded binary" ?

File: lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb [hunk 3]
```diff
@@ -160,14 +168,22 @@ def upload(app_id, binary_path, platform)
...
+      #
+      # args
+      #   app_id - Firebase App ID
+      #   upload_token - URL encoded upload token
+      #
+      # Returns the release_id on a successful release, otherwise returns nil.
```
Comment (lfkellogg) [inline lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb:177]: For documentation like this, I think referring to it in sentence case is preferred unless the identifier is meaningful somehow: "release ID"

File: lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb (hunk 2 above)
Comment (lfkellogg) [reply lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb:171]: I see you resolved this, but did you forget to push up a commit?

File: lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb (hunk 3 above)
Comment (lfkellogg) [reply lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb:177]: I see you resolved this, but did you forget to push up a commit?

Comment (alonsosalasinfante) [reply lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb]: Yup I did, pushed the new commit now

File: lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb [hunk 4]
```diff
@@ -110,9 +118,9 @@ def upload_binary(app_id, binary_path, platform)
           request.headers["X-APP-DISTRO-API-CLIENT-VERSION"] = Fastlane::FirebaseAppDistribution::VERSION
         end
       rescue Faraday::ResourceNotFound
-        UI.crash!("#{ErrorMessage::INVALID_APP_ID}: #{app_id}")
+        UI.user_error!("#{ErrorMessage::INVALID_APP_ID}: #{app_id}")
       rescue Errno::ENOENT
```
Comment (tagboola) [inline lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb:122]: nit: Not 100% clear that `Errno::ENOENT` is raised when the file can't be opened. Consider adding a comment here below to explain the reason for the exception being raised.

Comment (alonsosalasinfante) [reply lib/fastlane/plugin/firebase_app_distribution/client/firebase_app_distribution_api_client.rb:122]: Good point

**Pull request merged**